parsl/executors/workqueue/executor.py

```diff
@@ -0,0 +1,438 @@
+                    see_worker_output=False,
+                    port=50000,
+                    launch_cmd=None,
+                    data_dir=".",
+                    log_dir=None,
+                    full=False,
+                    password=None,
+                    password_file=None,
+                    project_name=None,
+                    env=None):
+
+    logger.debug("Starting WorkQueue Master Thread")
+
+    wq_to_parsl = {}
+    if log_dir is not None:
+        wq_debug_log = os.path.join(log_dir, "debug")
+        cctools_debug_flags_set("all")
+        cctools_debug_config_file(wq_debug_log)
+
+    logger.debug("Creating Workqueue Object")
+    try:
+        q = WorkQueue(port)
+    except Exception as e:
+        logger.error("Unable to create Workqueue object: {}", format(e))
+        raise e
+
+    if project_name:
+        q.specify_name(project_name)
+
+    if password:
+        q.specify_password(password)
+    elif password_file:
+        q.specify_password_file(password_file)
+
+    # Only write Logs when the log_dir is specified, which is most likely always will be
+    if log_dir is not None:
+        wq_master_log = os.path.join(log_dir, "master_log")
+        wq_trans_log = os.path.join(log_dir, "transaction_log")
+        if not full:
+            wq_resource_log = os.path.join(log_dir, "resource_logs")
+            q.enable_monitoring_full(dirname=wq_resource_log)
+
+        q.specify_log(wq_master_log)
+        q.specify_transactions_log(wq_trans_log)
+
+    continue_running = True
+    while(continue_running):
+        # Monitor the Task Queue
+
+        place_holder_queue = []
+
+        queue_lock.acquire()
+        queue_len = len(task_queue)
+        while queue_len > 0:
+            place_holder_queue.append(task_queue.pop())
+            queue_len -= 1
+        queue_lock.release()
+
+        # Submit Tasks
+        for item in place_holder_queue:
+            parsl_id = item["task_id"]
+            if parsl_id < 0:
+                continue_running = False
+                break
+            function_data_loc = item["data_loc"]
+            function_result_loc = item["result_loc"]
+            function_result_loc_remote = function_result_loc.split("/")[-1]
+            function_data_loc_remote = function_data_loc.split("/")[-1]
+
+            input_files = item["input_files"]
+            output_files = item["output_files"]
+
+            # TODO Make this general
+            # full_script_name = "/afs/crc.nd.edu/user/a/alitteke/parsl/parsl/executors/workqueue/workqueue_worker.py"
+            full_script_name = workqueue_worker.__file__
+            script_name = full_script_name.split("/")[-1]
+
+            # script_name = "workqueue_worker.py"
+            # full_script_name = shutil.which(script_name)
+            if full_script_name is None:
+                logger.error("Unable to find script {} on path, skipping task {}".format(script_name, parsl_id))
```

Comment:
Does this mean that the task is left hanging, not marked as failed at the parsl level?

Reply:
I believe that it is here, this line might not be needed anymore yet either.  But I do need some guidance about how to handle task failure in Parsl

Reply:
When you've got a successful result, you pass it back to parsl like this:
```
future.set_result(result)
```

If you want to indicate that the task has failed, instead set the future to an exception (the most python-like way):

```
future.set_exception(someexception)
```

or set the result using set_result to an instance of RemoteExceptionWrapper. This is a helper class that helps with some serialisation of exceptions from remote workers - so you might find it useful. Look in `executors/high_through/process_worker_pool.py` for an example of creating a RemoteExceptionWrapper.

Either of those methods for setting a failure will result in the exception you pass in popping out into the driving workflow script when future.result() is called.